### ISSUE TITLE
display: flex on help box removing space

### ISSUE THREAD

**Issue by rmjohnson** [Faithlife/styled-ui#197]
My JSX:
```jsx
<HelpBox showLightBulb>
	{
		'To modify a subscription that’s billed to someone else, contact the billed user or call Customer Service at '
	}
	<a href="tel:+18008756467">{'800-875-6467'}</a>
	{'.'}
</HelpBox>
```

Outputted HTML:
```html
<div class="styled__HelpBoxContent-sc-108ywmy-0 zMpIH">To modify a subscription that’s billed to someone else, contact the billed user or call Customer Service at <a href="tel:+18008756467">800-875-6467</a>.</div>
```

I've got a space and then an anchor tag inside a help box. The `display: flex` on `HelpBoxContent` makes it look like there is no space between "at" and the phone number. I'm not sure if this is me "doing it wrong" (I noted a `&nbsp` seems to fix it) but I'd expect this to work, especially in a component where we're likely to put text next to links.

![image](https://user-images.githubusercontent.com/249152/61012193-4605fa80-a332-11e9-9273-5743684df7e8.png)

**Comment by smaxa20** [comment]
After much Googling I've found a solution that doesn't involve `&nbsp` (and everything I saw said not to use `&nbsp` so that's a good thing).

Here is a great [stack overflow answer](https://stackoverflow.com/a/39334882/9968135) about exactly what you're seeing.

TL;DR: 
Inline elements keep the whitespace after/between elements, block elements don't, and flexbox makes all its children into block elements. The solution is to change the `white-space` property of the container to either `pre` or `pre-wrap`.

Here is some code that works as you want it to:
```
<HelpBox style={{ whiteSpace: 'pre' }} showLightBulb handleClose={() => true}>
	{'Hello '}
	<a href="https://www.google.com">Google</a>
</HelpBox>
```
![Capture3](https://user-images.githubusercontent.com/31995444/61072992-4ef5db00-a3c9-11e9-8ed6-884dbe64cad7.PNG)


I will look into whether or not we want to add this to the component and which value we want to use, but for now I think using either `whiteSpace: 'pre'` or `whiteSpace: 'pre-wrap'` is a better workaround than using a `&nbsp`.

**Comment by bryanrsmith** [comment]
Wrapping the contents with a container would be better than messing with whiteSpace. What are the downsides to the component adding a wrapper around children?

**Comment by smaxa20** [comment]
Nothing is necessarily wrong with adding a wrapper. But recently we've been taking out wrappers and overall avoiding adding more to the DOM, so I was continuing that trend.

This works just as well:
```html
<HelpBox showLightBulb>
	<span>
		{'Hello '}
		<a href="https://www.google.com">Google</a>
	</span>
</HelpBox>
```

**Comment by smaxa20** [comment]
Or even better you could use the `HelpBox.Body` component as a wrapper. It is already used as a wrapper whenever `HelpBox.Footer` is used.

**Comment by bryanrsmith** [comment]
> recently we've been taking out wrappers and overall avoiding adding more to the DOM

My goal isn't to reduce the number of DOM elements, it's to retain simplicity and flexibility in the consumer's ability to customize styling. In the case of the button, we had styling split between the button and the wrapper within it, making it very awkward to apply styling customizations to the button (which is generally used as a primitive and highly customizable element. In this case, there's probably no benefit to omitting the wrapper around the consumer's content because it's a more complex component that isn't intended to be very flexible in terms of styling customizability. It sounds like the omission of that wrapper is causing the component to behave surprisingly.

**Comment by smaxa20** [comment]
Ok that makes sense. 
> It sounds like the omission of that wrapper is causing the component to behave surprisingly.

Should I go and add a wrapper, or I can add a demo that showcases this use case?

**Comment by bryanrsmith** [comment]
I'm curious if there are any downsides to adding the wrapper. My preference would be for the component to behave unsurprisingly, and consistently. It sounds like the component providing the wrapper might be the best path there. If we start adding a wrapper only when a consumer specifies a footer, that seems both inconsistent and potentially surprising.

**Comment by smaxa20** [comment]
I can't think of any downsides really. 

Right now the body wrapper that we provide when the consumer specifies a footer doesn't fix this problem because it's a block element. If I'm not wrong, in order to allow for what Ryan needs we would need to add a span underneath that body wrapper, and a span in its place when that body wrapper isn't specified.

**Comment by RobertBolender** [comment]
I'm closing this issue since it hasn't had any activity since July, and there seems to be a reasonable solution with using a wrapping component inside the Helpbox.

Feel free to re-open this issue if there is still something that we should address or discuss.
